gradio/component.py

```diff
@@ -95,3 +111,233 @@ def get_all_shortcut_implementations(cls):
+        else:
+            raise ValueError(
+                "Unknown type: "
+                + str(self.type)
+                + ". Please choose from: 'str', 'number'."
+            )
+
+    def preprocess_example(self, x: str) -> str:
+        """
+        Returns:
+        (str): Text representing function input
+        """
+        return x
+
+    def set_interpret_parameters(
+        self, separator: str = " ", replacement: Optional[str] = None
+    ):
+        """
+        Calculates interpretation score of characters in input by splitting input into tokens, then using a "leave one out" method to calculate the score of each token by removing each token and measuring the delta of the output value.
+        Parameters:
+        separator (str): Separator to use to split input into tokens.
+        replacement (str): In the "leave one out" step, the text that the token should be replaced with. If None, the token is removed altogether.
+        """
+        self.interpretation_separator = separator
+        self.interpretation_replacement = replacement
+        return self
+
+    def tokenize(self, x: str) -> Tuple[List[str], List[str], None]:
+        """
+        Tokenizes an input string by dividing into "words" delimited by self.interpretation_separator
+        """
+        tokens = x.split(self.interpretation_separator)
+        leave_one_out_strings = []
+        for index in range(len(tokens)):
+            leave_one_out_set = list(tokens)
+            if self.interpretation_replacement is None:
+                leave_one_out_set.pop(index)
+            else:
+                leave_one_out_set[index] = self.interpretation_replacement
+            leave_one_out_strings.append(
+                self.interpretation_separator.join(leave_one_out_set)
+            )
+        return tokens, leave_one_out_strings, None
+
+    def get_masked_inputs(
+        self, tokens: List[str], binary_mask_matrix: List[List[int]]
+    ) -> List[str]:
+        """
+        Constructs partially-masked sentences for SHAP interpretation
+        """
+        masked_inputs = []
+        for binary_mask_vector in binary_mask_matrix:
+            masked_input = np.array(tokens)[np.array(binary_mask_vector, dtype=bool)]
+            masked_inputs.append(self.interpretation_separator.join(masked_input))
+        return masked_inputs
+
+    def get_interpretation_scores(
+        self, x, neighbors, scores: List[float], tokens: List[str], masks=None
+    ) -> List[Tuple[str, float]]:
+        """
+        Returns:
+        (List[Tuple[str, float]]): Each tuple set represents a set of characters and their corresponding interpretation score.
+        """
+        result = []
+        for token, score in zip(tokens, scores):
+            result.append((token, score))
+            result.append((self.interpretation_separator, 0))
+        return result
+
+    def generate_sample(self) -> str:
+        return "Hello World"
+
+    def postprocess(self, y: str):
```

Comment:
This is the main problem. The Textbox postprocess *needs* to be type aware.

```python
    def postprocess(self, y):
        """
        Parameters:
        y (str): text output
        Returns:
        (Union[str, number]): output value
        """
        if self.type == "str" or self.type == "auto":
            return str(y)
        elif self.type == "number":
            return y
        else:
            raise ValueError(
                "Unknown type: " + self.type + ". Please choose from: 'str', 'number'"
            )
```

Reply:
I am planning to keep breaking changes to the minimum. I don't think we need number type in the Textbox, because we will have a Number component both for the input and the output. People can use the Number component if they need it.

Reply:
And having it for the output case makes the design complexier and feels not that meaningful to me when we have a Number component.

Reply:
We don't have a `Number` output component right now. We could create one but we don't need to break existing code that uses `outputs.Textbox(type="number")`